pallets/pips/src/types.rs

```
@@ -43,26 +42,42 @@ pub const MAX_NORMAL_PRIORITY: Priority = HARD_DEADLINE + 1;
 pub(crate) const PIPS_LOCK_ID: LockIdentifier = *b"pips    ";
 
 /// A wrapper for a proposal description.
-#[derive(Decode, Encode, TypeInfo, VecU8StrongTyped)]
+#[derive(Decode, DecodeWithMemTracking, Encode, TypeInfo, VecU8StrongTyped)]
 #[derive(Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct PipDescription(pub Vec<u8>);
 
 /// The global and unique identitifer of a Polymesh Improvement Proposal (PIP).
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-#[derive(Decode, Encode, TypeInfo, MaxEncodedLen)]
+#[derive(Serialize, Deserialize)]
+#[derive(Decode, DecodeWithMemTracking, Encode, TypeInfo, MaxEncodedLen)]
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Default, Debug)]
 pub struct PipId(pub u32);
 impl_checked_inc!(PipId);
 
 impl PipId {
-    /// Converts a PIP ID into a name of a PIP scheduled for execution.
-    pub fn execution_name(&self) -> Vec<u8> {
-        (PIP_EXECUTION, self.0).encode()
+    /// Converts [`PipId`] into a [`TaskName`] of a PIP scheduled for execution.
+    pub fn execution_name(&self) -> Result<TaskName, Vec<u8>> {
+        let mut task_name: TaskName = [0; 32];
+        let encoded_task = (PIP_EXECUTION, self.0).encode();
+
+        if encoded_task.len() >= task_name.len() {
+            return Err(b"Task name too long".to_vec());
+        }
```

We could return the pallet Error type here, since this code is inside the pallet.  Not an important/required changed.